skll/data/readers.py

```diff
@@ -325,7 +332,7 @@ def _parse_dataframe(self, df, id_col, label_col, features=None):
         # delete the column; otherwise, just
         # set it to None
         if id_col is not None and id_col in df:
-            ids = df[id_col]
+            ids = df[id_col].astype(str)
```

Comment:
Can we not do this inside `_sub_read()` in `pd.read_csvmethod of `CSVReader` instead of doing it here. 

`df = pd.read_csv(file, sep=self._sep, engine=self._engine, dtype={self.id_col: str}, **self._pandas_kwargs)`

Reply:
I don't think we can because at that point `self.id_col` may be `None`. `_parse_dataframe` checks for that internally.

Reply:
I think it works okay
```
In [13]: ! head -1 ./tests/train/folds_file_test.csv
id,cv_fold_num

In [14]: df = pd.read_csv("./tests/train/folds_file_test.csv", dtype={None: float})

In [15]: df.columns
Out[15]: Index(['id', 'cv_fold_num'], dtype='object')
```

Reply:
But what's the advantage - there's no point in passing the argument `{None: float}` if you know it doesn't do anything?

Reply:
I also think it's smarter to do this inside the `_parse_dataframe` method, especially if we add other flat file readers (Excel, FWF) that rely on `pandas` in the future. We know that the IDs are always being forced to strings inside that method, regardless of what is passed to the underlying `pandas` reader.